modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/BoundObjectDefinitionsExportImportTest.java

```diff
@@ -0,0 +1,260 @@
+			_jsonFactory.createJSONArray(draftObjectDefinitionsJSONString);
+
+		for (Object objectDefinition : publishedObjectDefinitionsJSONArray) {
+			JSONObject objectDefinitionJSONObject =
+				(JSONObject)objectDefinition;
+
+			objectDefinitionJSONObject.put(
+				"status",
+				_jsonFactory.createJSONObject(
+				).put(
+					"code", 0
+				).put(
+					"label", "approved"
+				).put(
+					"label_i18n", "Approved"
+				));
+		}
+
+		testExportImportJSONString(
+			publishedObjectDefinitionsJSONArray.toString(),
+			draftObjectDefinitionsJSONString, null, "TestObjectDefinition1");
+
+		// Invalid bound object definitions
+
+		JSONArray invalidObjectDefinitionsJSONArray =
+			_jsonFactory.createJSONArray(draftObjectDefinitionsJSONString);
+
+		for (Object objectDefinition : invalidObjectDefinitionsJSONArray) {
+			JSONObject objectDefinitionJSONObject =
+				(JSONObject)objectDefinition;
+
+			objectDefinitionJSONObject.put(
+				"name", "!@" + objectDefinitionJSONObject.get("name"));
+		}
+
+		Class<?> clazz = getClass();
+
+		testFailedImportJSONString(
+			invalidObjectDefinitionsJSONArray.toString(),
+			StringUtil.read(
+				clazz.getResourceAsStream(
+					"dependencies/test-invalid-bound-object-definitions." +
+						"error-message.json")),
+			null, null);
+	}
+
+	@Override
+	protected ClassLoader getClassLoader() {
+		return BoundObjectDefinitionsExportImportTest.class.getClassLoader();
+	}
+
+	@Override
+	protected Class<?> getClazz() {
+		return getClass();
+	}
+
+	@Override
+	protected long getId(String name) throws Exception {
+		Page<ObjectDefinition> objectDefinitionsPage =
+			_objectDefinitionResource.getObjectDefinitionsPage(
+				name, null, null, Pagination.of(1, 1), null);
+
+		List<ObjectDefinition> objectDefinitions =
+			(List<ObjectDefinition>)objectDefinitionsPage.getItems();
+
+		ObjectDefinition objectDefinition = objectDefinitions.get(0);
+
+		return objectDefinition.getId();
+	}
+
+	@Override
+	protected String getIdentifierName() {
+		return "objectDefinitionId";
+	}
+
+	@Override
+	protected String getJSONName() {
+		return "objectDefinitionJSON";
+	}
+
+	@Override
+	protected MVCActionCommand getMVCActionCommand() {
+		return _mvcActionCommand;
+	}
+
+	@Override
+	protected MVCResourceCommand getMVCResourceCommand() {
+		return _mvcResourceCommand;
+	}
+
+	private JSONObject _createOneToManyObjectRelationship(
+			String objectDefinitionExternalReferenceCode1,
+			String objectDefinitionExternalReferenceCode2,
+			String objectDefinitionName2, String objectRelationshipName)
+		throws Exception {
+
+		return _jsonFactory.createJSONObject(
+			_defaultObjectRelationshipJSONString
+		).put(
+			"edge", true
+		).put(
+			"name", objectRelationshipName
+		).put(
+			"objectDefinitionExternalReferenceCode1",
+			objectDefinitionExternalReferenceCode1
+		).put(
+			"objectDefinitionExternalReferenceCode2",
+			objectDefinitionExternalReferenceCode2
+		).put(
+			"objectDefinitionName2", objectDefinitionName2
```

Comment:
Is this info required? objectDefinitionName2 is required for our tests? :thinking:

Reply:
If we don't specify it, it will fail in the json comparison